packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -38,12 +47,50 @@ function ListViewBlockSelectButton(
 	},
 	ref
 ) {
+	const { clientId } = block;
+	const inputRef = useRef();
+
+	// Setting managed via `toggleLabelEditingMode` handler.
+	const [ labelEditingMode, setLabelEditingMode ] = useState( false );
+
 	const blockInformation = useBlockDisplayInformation( clientId );
 	const blockTitle = useBlockDisplayTitle( {
 		clientId,
 		context: 'list-view',
 	} );
+
+	const [ inputValue, setInputValue ] = useState( blockTitle );
```

Comment:
Do we really need this state? Why not update the attribute directly when there are changes?
Right now, if I change the name and then click in another place, the changes I did are lost, I need to press enter for the changes to be applied. This behavior is different from all the other input fields, e.g., for any field on the inspector, I don't need to press enter for a chance to be applied. They are applied right away.

Reply:
> This behavior is different from all the other input fields, e.g., for any field on the inspector, I don't need to press enter for a chance to be applied. They are applied right away.

Hmmm that's a UX question. Yes I did this intentionally because I felt that you'd need to explicitly "commit" the result.

I'm sure I had some feedback which asked for this specifically but I can't seem to find it. It should be trivial to change if we think an "auto-commit" approach would be best.